### PR TITLE
MemoryMappedFile improvements

### PR DESCRIPTION
**Pull request by esdrubal**
MemoryMappedViewAccessor and MemoryMappedViewStream are now tested with
offsets.

Implemented MemoryMappedViewAccessor.PointerOffset and
MemoryMappedViewStream.PointerOffset.

Corlib now uses referencesource System.IO.UnmanagedMemoryAccessor.

### CODE REVIEW

File: mcs/class/corlib/System.Runtime.InteropServices/Marshal.cs [hunk 1]
```diff
@@ -979,6 +979,21 @@ public static int SizeOf (object structure)
 			return SizeOf (structure.GetType ());
 		}
 
+		internal static uint SizeOfType(Type type)
```
Comment (knocte) [inline mcs/class/corlib/System.Runtime.InteropServices/Marshal.cs]: missing space before parenthesis

File: mcs/class/corlib/System.Runtime.InteropServices/Marshal.cs [hunk 2]
```diff
@@ -979,6 +979,21 @@ public static int SizeOf (object structure)
...
+		{
+			return (uint) SizeOf (type);
+		}
+
+		internal static uint AlignedSizeOf<T>() where T : struct
```
Comment (knocte) [inline mcs/class/corlib/System.Runtime.InteropServices/Marshal.cs]: missing space before parenthesis

File: mcs/class/corlib/System.Runtime.InteropServices/Marshal.cs [hunk 3]
```diff
@@ -979,6 +979,21 @@ public static int SizeOf (object structure)
...
+		{
+			uint size = SizeOfType(typeof(T));
```
Comment (knocte) [inline mcs/class/corlib/System.Runtime.InteropServices/Marshal.cs]: missing space before parenthesis (twice)

**Comment by akoeplinger** [comment]
There's a typo in this PR that breaks the build, see: https://github.com/mono/mono/pull/1854

**Pull request merged**